### DISCUSSION TITLE
Updating config file with new features

### DISCUSSION THREAD

**Discussion by nebulosa2007** [YaLTeR/niri#152] [Ideas]
There should also be an update mechanism for adding new lines and features to the user's configuration file. Otherwise, Niri use default config newer than the user has and user not warned about it.

Since this [commit](https://github.com/YaLTeR/niri/commit/51243a0a505a533057e7326fbbae882420f0d363)

**Comment by YaLTeR** [comment]
When I add new config options I make sure the previous configs without them will have them at the default values, so it shouldn't be a big issue. I try to document all of them in release notes.

**Comment by nebulosa2007** [comment]
I'm not talking about default values and how Niri's work with them, but about how edit default values.

For example, this [commit](https://github.com/YaLTeR/niri/blob/962e159db61dc6c7822aa899b1d9dc86fb6a0de5/resources/default-config.kdl#L68), which will allow to change screen rotation. All new options need to be moved manually, right?

**Reply by YaLTeR** [reply]
Wdym? If you don't need rotation, you just do nothing and it keeps working. If you need rotation, you write `transform "90"` or similar.

**Reply by nebulosa2007** [reply]
I mean something like:

`git diff ~/.config/niri/config.kdl /usr/share/doc/niri/default-config.kdl --word-diff | grep -P '{\+|\[\-' --color=never`

Then show those lines or write in ~/.config/niri/config.kdl in proper order.

**Reply by YaLTeR** [reply]
Well, that's kinda complicated to do right. Plus, I'm sure there are people who'd rather their config not touched every update

**Reply by nebulosa2007** [reply]
Almost all day think how it could be, yes it's complicated. 

Maybe write some instruction in the wiki - on a new release: move config.klm to config.klm.backup, reload Niri and manually edit config file. Something like that. 
 I understand that if the option is not activated in the config file, Niri use the default settings, but still somehow the config file needs to be updated periodically.
 
 Not a big deal, of course, just thoughts.

**Comment by kchibisov** [comment]
It's usually solved with putting the actual docs into separate place, like man page.

**Comment by 0xDEAD-C0DE** [comment]
I agree with @kchibisov about new features documented in a man page. User configuration should not be touch at all.

**Comment by nebulosa2007** [comment]
Meanwhile, I made simple script for updating main config:

`    Mod+Shift+Ctrl+Slash { spawn "bash" "-c" "$HOME/.config/niri/update_niri_config"; }`

and 

```
#!/bin/bash

url="https://raw.githubusercontent.com/YaLTeR/niri/main/resources/default-config.kdl"
config_user="$HOME/.config/niri/config.kdl"
config_default_niri="/tmp/default-config.kdl"

curl -sLR -o $config_default_niri $url
meld $config_default_niri $config_user
rm $config_default_niri
```
